spring-cloud-gcp-integration/src/main/java/org/springframework/integration/gcp/storage/GcsFileInfo.java

```diff
@@ -60,7 +60,8 @@ public String getFilename() {
 
 	@Override
 	public String getPermissions() {
-		throw new UnsupportedOperationException("Use [BlobInfo.getAcl()] to obtain permissions.");
+		// throw new UnsupportedOperationException("Use [BlobInfo.getAcl()] to obtain permissions.");
+		return "xyz";
```

Comment:
What is this, please?

Reply:
@artembilan I meant to ask you about this.

`getPermissions()` is somehow being called by our tests with SF5. Do you know what's going on? I'm not sure what the best way forward here is.

Reply:
OK. I see now. The super class has a `toString()` with the ` getPermissions()` call as well.
In SI-AWS I have this override:
```
public String toString() {
	return "FileInfo [isDirectory=" + isDirectory() + ", isLink=" + isLink()
			+ ", Size=" + getSize() + ", ModifiedTime="
			+ new Date(getModified()) + ", Filename=" + getFilename()
			+ ", RemoteDirectory=" + getRemoteDirectory() + "]";
}
```
And that's all!

Reply:
@artembilan I tried that suggestion, but it isn't working for me.

The issue is that on the storage inbound streaming adapter test, the payload is being transformed to JSON and the `permissions` descriptor is invoking the `getPermissions()` file directly.

Need to dig on this a bit more. But apart from this issue, and the fact that payloads are now being sent on the Stream binder as `application/json` instead of `text/plain`, I think we're pretty much worry free with this migration.

Reply:
For posterity, there is another issue with the `netty-tcnative-boringssl-static` dependency.

```Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.netty.handler.ssl.OpenSslEngine```

`google-cloud-java` mandates version `2.0.6.Final` which is incompatible with netty version `4.1.17.Final`.
We might need to add `netty-tcnative-boringssl-static`'s version to our bom.

```
		<dependency>
			<groupId>io.netty</groupId>
			<artifactId>netty-tcnative-boringssl-static</artifactId>
			<version>2.0.7.Final</version>
		</dependency>
```

Reply:
Ah! I think this one: https://github.com/spring-projects/spring-integration/pull/2308.
This is in the SI-5.0.1.BUILD-SNAPSHOT, so, we might need to add `<spring-integration.version>5.0.1.BUILD-SNAPSHOT</spring-integration.version>` to our pom for time being.

> google-cloud-java mandates version 2.0.6.Final

That's bad. That's not a modern world. It must be upgraded there for the actual version otherwise it really makes it useless to use `google-cloud-java` with Spring...

Reply:
Set the SI version to `5.0.1.BUILD-SNAPSHOT` and it solved our issue.

google-cloud-java are definitely setting the version for that dependency. https://github.com/GoogleCloudPlatform/google-cloud-java/blob/184c219c7a7d38bd8146e3a4d35f0972a1d24a10/google-cloud-bom/pom.xml#L136

I'll ask them if they can remove it from their BOM.
For the moment, it seems like we'll have to live with setting the version on our side?

Reply:
https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2781

In the meantime, for the `netty-tcnative-boringssl-static` issue, sounds like we either set the version on our side, or ask users to set it on their side.